internal/registry/strategy_registry.go

```diff
@@ -0,0 +1,71 @@
+package registry
+
+import (
+	"sync"
+
+	"github.com/cloudflare/tf-migrate/internal/interfaces"
+)
+
+type StrategyRegistry struct {
+	strategies map[string]interfaces.ResourceTransformer
+	order      []string
+	mu         sync.RWMutex
+}
+
+func NewStrategyRegistry() *StrategyRegistry {
+	return &StrategyRegistry{
+		strategies: make(map[string]interfaces.ResourceTransformer),
+		order:      make([]string, 0),
+	}
+}
+
+func (r *StrategyRegistry) Register(strategy interfaces.ResourceTransformer) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	resourceType := strategy.GetResourceType()
+	r.strategies[resourceType] = strategy
+	r.order = append(r.order, resourceType)
+}
+
+func (r *StrategyRegistry) Find(resourceType string) interfaces.ResourceTransformer {
```

Comment:
Why is this func needed? Where does it get used or what is its use case?

Edit: same question for all of the methods on `StrategyRegistry`.